### DISCUSSION TITLE
Keep all audio and subtitles?

### DISCUSSION THREAD

**Discussion by Chris230291** [streamlink/streamlink#5426] [Q&A] [answered]
### Description

Is there an equivalent of these ffmpeg commands?

 `-map 0:a?`
 `-map 0:s?`

I saw the hls option, but what If the stream isn't hls?

**Comment by bastimeyer** [comment, marked as answer]
Streamlink doesn't support muxing multiple video/audio/subtitle streams. Only one substream of each kind gets chosen from the available streams (depending on the protocol) and they then muxed into the output. This is what the stream selection is for, which is the cartesian product of the available streams in most cases when the output is a subclass of MuxedStream.

**Reply by Chris230291** [reply]
Ok thank you.
